app/src/main/java/de/rochefort/childmonitor/ListenService.kt

```diff
@@ -0,0 +1,212 @@
+/*
+ * This file is part of Child Monitor.
+ *
+ * Child Monitor is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Child Monitor is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Child Monitor. If not, see <http://www.gnu.org/licenses/>.
+ */
+package de.rochefort.childmonitor
+
+import android.app.Notification
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.app.Service
+import android.content.Intent
+import android.content.pm.ServiceInfo
+import android.media.AudioManager
+import android.media.AudioTrack
+import android.media.MediaPlayer
+import android.os.Binder
+import android.os.Build
+import android.os.IBinder
+import android.util.Log
+import android.widget.Toast
+import androidx.core.app.NotificationCompat
+import androidx.core.app.ServiceCompat
+import java.io.IOException
+import java.net.Socket
+
+class ListenService : Service() {
+    private val frequency: Int = AudioCodecDefines.Companion.FREQUENCY
+    private val channelConfiguration: Int = AudioCodecDefines.Companion.CHANNEL_CONFIGURATION_OUT
+    private val audioEncoding: Int = AudioCodecDefines.Companion.ENCODING
+    private val bufferSize = AudioTrack.getMinBufferSize(frequency, channelConfiguration, audioEncoding)
+    private val byteBufferSize = bufferSize * 2
+    private val binder: IBinder = ListenBinder()
+    private var notificationManager: NotificationManager? = null
+    private var listenThread: Thread? = null
+    val volumeHistory = VolumeHistory(16384)
+    var childDeviceName: String? = null
+        private set
+
+    override fun onCreate() {
+        super.onCreate()
+        this.notificationManager = getSystemService(NOTIFICATION_SERVICE) as NotificationManager
+    }
+
+    override fun onStartCommand(intent: Intent, flags: Int, startId: Int): Int {
+        Log.i(TAG, "Received start id $startId: $intent")
+        // Display a notification about us starting.  We put an icon in the status bar.
+        createNotificationChannel()
+        val extras = intent.extras
+        if (extras != null) {
```

Comment:
use intent.extras?.let instead of the null check here

Reply:
Done (and other places where I did use that pattern).